### PR TITLE
two handlers and a better test

### PR DESCRIPTION
**Pull request by mikedewar**

### CODE REVIEW

File: server/source.go [hunk 1]
```diff
@@ -62,7 +62,30 @@ func (s *Server) SourceIndexHandler(w http.ResponseWriter, r *http.Request) {
...
+	}
+	s.Lock()
+	defer s.Unlock()
+	source, ok := s.sources[id]
+	if !ok {
+		w.WriteHeader(http.StatusBadRequest)
+		writeJSON(w, Error{"could not find source"})
+		return
+	}
+	writeJSON(w, source)
```
Comment (nikhan) [inline server/source.go]: add w.WriteHeader(http.StatusOK) please

File: server/group.go [hunk 2]
```diff
@@ -270,8 +270,33 @@ func (s *Server) GroupDeleteHandler(w http.ResponseWriter, r *http.Request) {
...
+	}
+	s.Lock()
+	defer s.Unlock()
+	group, ok := s.groups[id]
+	if !ok {
+		w.WriteHeader(http.StatusBadRequest)
+		writeJSON(w, Error{"could not find group"})
+		return
+	}
+	writeJSON(w, group)
```
Comment (nikhan) [inline server/group.go]: add w.WriteHeader(http.StatusOK) please

**Comment by mikedewar** [comment]
added 200 headers.

**Comment by nikhan** [comment]
:cat: :cactus:

**Pull request merged**